packages/calcite-components/src/components/list/list.tsx

```diff
@@ -744,7 +761,7 @@ export class List
     const filteredItems = filterPredicate
```

Comment:
The way this is currently structured, filterPredicate would only affect items and not groups.

Reply:
Meaning to ask, should the `filterPredicate` include groups?  It wasn't mentioned here https://github.com/Esri/calcite-design-system/issues/3454 but (https://github.com/Esri/calcite-design-system/issues/7702) was created later.

Reply:
I don't think it should, just wanted to point out the inconsistency with treating groups like items.

Reply:
If not, is the alternative `metadata` + `filterProps`? While functional, it introduces boilerplate for a common filtering workflow using information already present in the list.